src/main/java/edu/cornell/kfs/tax/dataaccess/impl/TransactionRowPRNCBuilder.java

```diff
@@ -0,0 +1,544 @@
+                    if (summary.stateTaxWithheldObjectCodes.contains(financialObjectCode)) {
+                        // If the object code represents a state-tax-withheld one, then use the non-reportable income code.
+                        incomeCode = summary.nonReportableIncomeCode;
+                        numDeterminedStateIncomeTaxWithheldIncomeCodes++;
+                    } else {
+                        numUndeterminedStateIncomeTaxWithheldIncomeCodes++;
+                        // If not a state-tax-withheld one either, then use the "excluded" income code.
+                        incomeCode = summary.excludedIncomeCode;
+                        numExcludedAssignedIncomeCodes++;
+                    }
+                }
+            }
+            
+            incomeCodeSubType = summary.incomeClassCodeToIrsIncomeCodeSubTypeMap.get(incomeClassCode);
+            if (StringUtils.isBlank(incomeCodeSubType)) {
+                // If no subtype was found, then set it to the exclusion value.
+                incomeCodeSubType = summary.excludedIncomeCodeSubType;
+                numExcludedAssignedIncomeCodeSubTypes++;
+            }
+            
+            // Prepare to update income code and income code subtype fields.
+            rs.updateString(summary.transactionDetailRow.incomeCode.index, incomeCode);
+            rs.updateString(summary.transactionDetailRow.incomeCodeSubType.index, incomeCodeSubType);
+        }
+        
+        @Override
+        EnumMap<TaxStatType,Integer> getStatistics() {
+            EnumMap<TaxStatType,Integer> stats = super.getStatistics();
+            
+            stats.put(TaxStatType.NUM_EXCLUDED_ASSIGNED_INCOME_CODES, Integer.valueOf(numExcludedAssignedIncomeCodes));
+            stats.put(TaxStatType.NUM_EXCLUDED_ASSIGNED_INCOME_CODES_DV, Integer.valueOf(numExcludedAssignedIncomeCodes));
+            stats.put(TaxStatType.NUM_EXCLUDED_ASSIGNED_INCOME_CODE_SUBTYPES, Integer.valueOf(numExcludedAssignedIncomeCodeSubTypes));
+            stats.put(TaxStatType.NUM_EXCLUDED_ASSIGNED_INCOME_CODE_SUBTYPES_DV, Integer.valueOf(numExcludedAssignedIncomeCodeSubTypes));
+            stats.put(TaxStatType.NUM_DETERMINED_FED_TAX_WITHHELD_INCOME_CODES, Integer.valueOf(numDeterminedFederalTaxWithheldIncomeCodes));
+            stats.put(TaxStatType.NUM_DETERMINED_FED_TAX_WITHHELD_INCOME_CODES_DV, Integer.valueOf(numDeterminedFederalTaxWithheldIncomeCodes));
+            stats.put(TaxStatType.NUM_UNDETERMINED_FED_TAX_WITHHELD_INCOME_CODES, Integer.valueOf(numUndeterminedFederalTaxWithheldIncomeCodes));
+            stats.put(TaxStatType.NUM_DETERMINED_STATE_INC_TAX_WITHHELD_INCOME_CODES, Integer.valueOf(numDeterminedStateIncomeTaxWithheldIncomeCodes));
+            stats.put(TaxStatType.NUM_DETERMINED_STATE_INC_TAX_WITHHELD_INCOME_CODES_DV, Integer.valueOf(numDeterminedStateIncomeTaxWithheldIncomeCodes));
```

Comment:
The DV enum constants are being referenced here instead of the PRNC ones. The "_PRNC"-suffixed ones should be used instead.